### PR TITLE
{numlib}[gompi/2017a,intel/2017a] FFTW v3.3.6, FFTW v3.3.6

### PR DESCRIPTION
**Pull request by akesandgren**
(created using `eb --new-pr`)
Update FFTW 3.3.6 from upstream git to fix MPI F03 interface.

### CODE REVIEW

File: easybuild/easyconfigs/f/FFTW/FFTW-3.3.6-intel-2017a.eb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    'FFTW-3.3.6-F03_interface_pl2.patch',
+]
+
+# no quad precision, requires GCC v4.6 or higher
+# see also
+# http://www.fftw.org/doc/Extended-and-quadruple-precision-in-Fortran.html
+with_quad_prec = False
+
+# compilation fails when configuring with --enable-avx-128-fma
+use_fma = False
```
Comment (boegel) [inline easybuild/easyconfigs/f/FFTW/FFTW-3.3.6-intel-2017a.eb]: @akesandgren Did you check whether this is still required?

File: easybuild/easyconfigs/f/FFTW/FFTW-3.3.6-intel-2017a.eb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+checksums = ['682a0e78d6966ca37c7446d4ab4cc2a1']
+
+patches = [
+    'FFTW-3.3.6-F03_interface_pl2.patch',
+]
+
+# no quad precision, requires GCC v4.6 or higher
+# see also
+# http://www.fftw.org/doc/Extended-and-quadruple-precision-in-Fortran.html
+with_quad_prec = False
```
Comment (boegel) [inline easybuild/easyconfigs/f/FFTW/FFTW-3.3.6-intel-2017a.eb]: Still not supported in this version of Intel compilers?

File: easybuild/easyconfigs/f/FFTW/FFTW-3.3.6-F03_interface_pl2.patch [hunk 3]
```diff
@@ -0,0 +1,1527 @@
...
+# The scripts were using fftw3.h from /usr/include, not ../api, and were
+# failing silently if fftw3.h was not installed.  This bug led to a
+# fftw-3.3.6pl1 release with incomplete mpi/f03 header files.
+# See commit 83092f8efbf872aefe7cfc6ee8fa43412f8e167a in FFTW upstream
+# git.
+#
+# This patch is the result of applying that commit and running the
+# maintainer mode scripts.
+#
+# Åke Sandgrem 2017-03-07
```
Comment (boegel) [inline easybuild/easyconfigs/f/FFTW/FFTW-3.3.6-F03_interface_pl2.patch]: @akesandgren typo ;)

File: easybuild/easyconfigs/f/FFTW/FFTW-3.3.6-F03_interface_pl2.patch [hunk 4]
```diff
@@ -0,0 +1,1527 @@
+# The scrips that generate the MPI F03 interface was broken in 3.3.6-pl1
+# The scripts were using fftw3.h from /usr/include, not ../api, and were
+# failing silently if fftw3.h was not installed.  This bug led to a
+# fftw-3.3.6pl1 release with incomplete mpi/f03 header files.
+# See commit 83092f8efbf872aefe7cfc6ee8fa43412f8e167a in FFTW upstream
+# git.
```
Comment (boegel) [inline easybuild/easyconfigs/f/FFTW/FFTW-3.3.6-F03_interface_pl2.patch]: include a URL to the commit?

**Comment by boegel** [comment]
@akesandgren No patch release for FFTW 3.3.6 coming up that includes this fix?

**Comment by boegel** [comment]
(close/open to wake up Travis)

@akesandgren I hope to include this in EasyBuild v3.1.2...

File: easybuild/easyconfigs/f/FFTW/FFTW-3.3.6-intel-2017a.eb (hunk 2 above)
Comment (akesandgren) [inline easybuild/easyconfigs/f/FFTW/FFTW-3.3.6-intel-2017a.eb]: It failed when i tried it, and there is still this issue, https://github.com/FFTW/fftw3/issues/83.

File: easybuild/easyconfigs/f/FFTW/FFTW-3.3.6-intel-2017a.eb (hunk 1 above)
Comment (akesandgren) [inline easybuild/easyconfigs/f/FFTW/FFTW-3.3.6-intel-2017a.eb]: It failed when i tried to use it, i.e., when i tried use_fma=true.
Haven't seen an issue about it in fftw issues though.

**Comment by akesandgren** [comment]
No, still no -pl2 release on their web, -pl2 is just a tag so far. Been reloading that page frequently, still nothing this morning.

**Comment by boegel** [comment]
@akesandgren Do you think it's worth the wait, or to contact the FFTW developers to ask for an ETA?

**Comment by akesandgren** [comment]
Don't know. Waiting is not an option for anyone that needs the F03 interface. (Unless they can postpone the storm of users complaining :-)

I've sent a mail to them...

**Comment by boegel** [comment]
@akesandgren Well, with your patch the immediate problem is solved, but I'd rather stick to an official release in the central easyconfigs repository if we have the option... Seems like it may be a matter of hours since the release is tagged already?

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
node2158.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/8b513e0259bcd943966db88cdb2b8292 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2635.swalot.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/37003aaf0b6d8e085d0da95809476614 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2157.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/7e959b8c5c4404eeb240749e8e3451a4 for a full test report.

**Comment by boegel** [comment]
Since the -pl2 FFTW 3.3.6 was tagged weeks ago at https://github.com/FFTW/fftw3/releases/tag/fftw-3.3.6-pl2, I'm not going to hold this back any further, so we can include it in EasyBuild v3.1.2

Thanks @akesandgren!

**Pull request merged**